viewing_party/party.py

```diff
@@ -1,23 +1,127 @@
+
+# --------- UTILITY FUNCTIONS -------------
+
+def make_list_unique(list_of_dicts):
+    #returns a list that has no duplicate dicts
+    unique_list = []
+    for dct in list_of_dicts:
+        if dct not in unique_list:
+            unique_list.append(dct)
+    return unique_list
+
+def get_friends_watched_movies(user_data):
+    friends_watched_movies = []
+    for dct in user_data["friends"]:
+        for movie in dct["watched"]:
+            friends_watched_movies.append(movie)
+    return friends_watched_movies
+
+
+# -----------------------------------------
 # ------------- WAVE 1 --------------------
+# -----------------------------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if (title is None) or (genre is None) or (rating is None):
+        return None
+    return {"title":title, "genre":genre, "rating":rating,}
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, movie_title):
+    '''
+    Cycles through watchlist, finds matching entry, then pops that
+    entry off of watchlist and appends it to watched.
+    '''
+    for i, movie in enumerate(user_data["watchlist"]):
+        if movie["title"] == movie_title:
+            user_data["watched"].append(user_data["watchlist"].pop(i))
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    ratings_list = [movie["rating"] for movie in user_data["watched"]]
+    if len(ratings_list) == 0:
+        return 0
+    else:
+        return sum(ratings_list) / len(ratings_list)
+

```

Comment:
Great job avoiding a divide by zero! Small note, the `else` is unneeded here because the function ends if there is a return.